persist/src/main/java/com.codeforcommunity/database/tableImpl/CommentTableDBImpl.java

```diff
@@ -0,0 +1,178 @@
+      // The '?' allows us to safely insert that variable into the query without having to worry
+      // about escaping any special characters inside.
+      String sql = "SELECT * FROM comments WHERE post_id = ?;";
+      // A PreparedStatement is the technique that allows us to insert variables by '?'.
+      PreparedStatement stmt = conn.prepareStatement(sql);
+      // Set the first '?' = id. Note how in prepared statements, parameters are not 0-indexed.
+      stmt.setInt(1, postId);
+
+      // Get our results. This could also be done in two separate calls;
+      // stmt.execute() and stmt.getResultSet().
+      ResultSet res = stmt.executeQuery();
+      // The next row in the table is queued up by calling ResultSet.next(). If ResultSet.next()
+      // returns false, then there are no more rows (or no rows were found if
+      // this is the first call).
+      while (res.next()) {
+        comments.add(allFieldsResultSetToRecord(res));
+      }
+
+      res.close();
+      stmt.close();
+      conn.close();
+    } catch (SQLException e) {
+      throw new IllegalStateException("There was an issue interacting with the database.", e);
+    }
+    return comments;
+  }
+
+  @Override
+  public void saveComment(CommentRecord comment) {
+    try {
+      Connection conn = getConnection();
+      String sql = "INSERT INTO comments (post_id, author, body) VALUES (?, ?, ?);";
+
+      PreparedStatement stmt = conn.prepareStatement(sql);
+      stmt.setInt(1, comment.getPostId());
+      stmt.setString(2, comment.getAuthor());
+      stmt.setString(3, comment.getBody());
+
+      stmt.execute();
+
+      stmt.close();
+      conn.close();
+    } catch (SQLException e) {
+      throw new IllegalStateException("There was an issue interacting with the database.", e);
+    }
+  }
+
+  @Override
+  public boolean commentExists(int postId, int commentId) {
+    boolean commentExists = false;
+    try {
+      Connection conn = getConnection();
+      String sql = "SELECT id FROM comments WHERE post_id = ? AND id = ?;";
+
+      PreparedStatement stmt = conn.prepareStatement(sql);
+      stmt.setInt(1, postId);
+      stmt.setInt(2, commentId);
+
+      ResultSet res = stmt.executeQuery();
+      if (res.next()) {
+        commentExists = true;
+      }
+
+      res.close();
+      stmt.close();
+      conn.close();
+    } catch (SQLException e) {
+      throw new IllegalStateException("There was an issue interacting with the database.", e);
+    }
+    return commentExists;
+  }
+
+  @Override
+  public void clapComment(int postId, int commentId) {
+    try {
+      Connection conn = getConnection();
+      // Here, we're updating the rows in the post table by the given id
+      // by incrementing the clap count.
+      String sql = "UPDATE comments SET clap_count = clap_count + 1 WHERE post_id = ? AND id = ?;";
+      PreparedStatement stmt = conn.prepareStatement(sql);
```

Comment:
So are we not having an option to loose a clap?

Reply:
Yeah, we're only incrementing because we don't have session information.